docs/stdlib/bool.rst

```diff
@@ -146,14 +144,13 @@ The truth tables are as follows:
 
 
 It is important to understand the difference between using
-``and``/``or`` vs :eql:func:`all`/:eql:func:`any`. This difference is
+``and``/``or`` and :eql:func:`all`/:eql:func:`any`. The difference is
 in how they handle ``{}``. Both ``and`` and ``or`` operators apply to
-the cross-product of their operands. Thus if any of the operands are
-``{}``, the result is also ``{}`` for ``and`` and ``or``.
+the cross-product of their operands. Thus, if any of the operands are
+``{}``, the result will be that.
```

Comment:
I'm finding this tough to parse. When I read it, I thought it was telling me that the difference in `and`/`or` and `all`/`any` was in the way they handle an empty set. When I tried it to confirm that, I realized that couldn't be what it was saying because they all throw when given an empty set.

Suggestion:
> \`\`and\`\`/\`\`or\`\`  and :eql:func:\`all\`/:eql:func:\`any\` differ in the way they handle sets of boolean values (e.g., \`\`{true, false}\`\`). Both \`\`and\`\` and \`\`or\`\` operators apply to the cross-product of their operands. If any of the operands is a set, the result will also be a set. For example:
>
> .. code-block:: edgeql-repl
>
>        db> select {true, false} and {true, false};
>        {true, false, false, false}
>        db> select {true, false} and true;
>        {true, false}

Doesn't have to happen in this PR, but I wanted to call it out here since part of the goal is to reword.

Reply:
This probably also results in some changes in the following paragraphs, unless I'm just horribly mistaken about the intent here.

Reply:
Good catch, we can make this part of the PR, this stuff should definitely be in

Reply:
@i0bs I think @vpetrovykh is going to check my understanding on this one, so you might want to hold off implementing it until he can take a look.

Reply:
The point here is indeed about differences in handling empty sets. The binary operators `and` and `or` must have both operands present to produce a value. Basically the way they work with sets is that they make a cross-product of all input sets and then they get applied tot he resulting cross product. This necessarily means that if any of the inputs are `{}`, the result is also `{}`. Conversely, the `any`/`all` functions are generalizations of the boolean operators to sets (in the same way as `sum` is generalization of `+` when applied to arbitrary sets). The `any`/`all` functions have well-defined behaviour even for `{}` input.

Reply:
This makes sense, @vpetrovykh. Thank you for clarifying. In that case, a new suggested rewrite:

```rest
``and``/``or`` and :eql:func:`all`/:eql:func:`any` differ in the way they
handle an empty set (``{}``). Both ``and`` and ``or`` operators apply to the
cross-product of their operands. If either operand is an empty set, the result
will also be an empty set. For example:

.. code-block:: edgeql-repl

    db> select {true, false} and <bool>{};
    {}
    db> select true and <bool>{};
    {}

Operating on an empty set with :eql:func:`all`/:eql:func:`any` does *not* return
an empty set:

.. code-block:: edgeql-repl

    db> select all(<bool>{});
    {true}
    db> select any(<bool>{});
    {false}

:eql:func:`all` returns ``true`` because the empty set contains no false values.
:eql:func:`any` returns ``false`` because the empty set contains no true
values.
```

@vpetrovykh Would you have time to read this over either bless it or recommend some changes? Happy to jump on a call if that's easier.

@i0bs Please hold off on this one until @vpetrovykh can take a look. I think we're getting there though.

Thanks, everyone!